lib/improver/nowcasting/utilities.py

```diff
@@ -0,0 +1,211 @@
+            oe_cubes (iris.cube.Cube or iris.cube.CubeList):
+                Cube or CubeList containing the orographic enhancement fields.
+
+        Returns:
+            oe_cube (iris.cube.Cube):
+                Cube containing the orographic enhancement fields at the
+                required time.
+
+        """
+        time_point, = iris_time_to_datetime(precip_cube.coord("time").copy())
+        oe_cube = extract_nearest_time_point(oe_cubes, time_point)
+        return oe_cube
+
+    def _apply_orographic_enhancement(self, precip_cube, oe_cube):
+        """Combine the precipitation rate cube and the orographic enhancement
+        cube.
+
+        Args:
+            precip_cube (iris.cube.Cube):
+                Cube containing the input precipitation field.
+            oe_cube (iris.cube.Cube):
+                Cube containing the orographic enhancement field matching
+                the validity time of the precipitation cube.
+
+        Returns:
+            cube (iris.cube.Cube):
+                Cube containing the precipitation rate field modified by the
+                orographic enhancement cube.
+
+        """
+        # Ensure the orographic enhancement cube matches the
+        # dimensions of the precip_cube.
+        oe_cube = check_cube_coordinates(precip_cube, oe_cube)
+
+        # Ensure that orographic enhancement is in the units of the
+        # precipitation rate cube.
+        oe_cube.convert_units(precip_cube.units)
+
+        # Set orographic enhancement to be zero for points with a
+        # precipitation rate of < 1/32 mm/hr.
+        original_units = Unit("mm/hr")
+        threshold_in_cube_units = (
+            original_units.convert(self.min_precip_rate_mmh,
+                                   precip_cube.units))
+        oe_cube.data[precip_cube.data < threshold_in_cube_units] = 0.
+
+        # Use CubeCombiner to combine the cubes.
+        temp_cubelist = iris.cube.CubeList([precip_cube, oe_cube])
+        cube = CubeCombiner(self.operation).process(
+            temp_cubelist, precip_cube.name())
+        return cube
+
+    def _apply_minimum_precip_rate(self, precip_cube, cube):
+        """Ensure that negative precipitation rates are capped at +1/32 mm/hr.
+
+        Args:
+            precip_cube (iris.cube.Cube):
+                Cube containing a precipitation rate input field.
+            cube (iris.cube.Cube):
+                Cube containing the precipitation rate field after combining
+                with orographic enhancement.
+
+        Returns:
+            cube (iris.cube.Cube):
+                Cube containing the precipitation rate field where any
+                negative precipitation rates have been capped at +1/32 mm/hr.
+
+        """
+        if self.operation == "subtract":
+            original_units = Unit("mm/hr")
+            threshold_in_cube_units = (
+                original_units.convert(self.min_precip_rate_mmh,
+                                       cube.units))
```

Comment:
`precip_cube.units`?  They should be the same though, so maybe this doesn't matter.

Reply:
I've made an edit just in case the cubes would have different units and added a unit test.